### PR TITLE
WebXR bindings for WebGPU

### PR DESCRIPTION
**Pull request by mwyrzykowski**
#### db28bb1f466a0df1f97a2b4d4eca431f21f24c6e
<pre>
WebXR bindings for WebGPU
<a href="https://bugs.webkit.org/show_bug.cgi?id=277727">https://bugs.webkit.org/show_bug.cgi?id=277727</a>
<a href="https://rdar.apple.com/133349967">rdar://133349967</a>

Reviewed by Tadeu Zagallo.

Add idl files and empty class implementations for WebGPU bindings for WebXR.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.h:
* Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl:
* Source/WebCore/Modules/webxr/XRGPUBinding.cpp: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl.
(WebCore::XRWebGLBinding::create):
* Source/WebCore/Modules/webxr/XRGPUBinding.h: Added.
(WebCore::XRGPUBinding::nativeProjectionScaleFactor const):
* Source/WebCore/Modules/webxr/XRGPUBinding.idl: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.h.
* Source/WebCore/Modules/webxr/XRGPULayerInit.h: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.h.
* Source/WebCore/Modules/webxr/XRGPULayerInit.idl: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl.
* Source/WebCore/Modules/webxr/XRGPUProjectionLayerInit.h: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.h.
* Source/WebCore/Modules/webxr/XRGPUProjectionLayerInit.idl: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl.
* Source/WebCore/Modules/webxr/XRGPUSubImage.cpp: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl.
* Source/WebCore/Modules/webxr/XRGPUSubImage.h: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.h.
(WebCore::XRGPUSubImage::colorTexture const):
(WebCore::XRGPUSubImage::depthStencilTexture const):
(WebCore::XRGPUSubImage::motionVectorTexture const):
* Source/WebCore/Modules/webxr/XRGPUSubImage.idl: Copied from Source/WebCore/Modules/WebGPU/GPURequestAdapterOptions.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:

Canonical link: <a href="https://commits.webkit.org/282270@main">https://commits.webkit.org/282270@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/d20537604c2efea42dd80490b11a95dd86587279

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62601 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41956 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15196 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66585 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13153 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64721 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13489 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50538 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9078 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65670 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38995 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54222 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31187 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/62112 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35717 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11557 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12081 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/55698 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57304 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11873 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68316 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/61844 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6547 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11556 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57816 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54269 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58015 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13903 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5465 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/83607 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37756 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14692 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38841 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39938 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38585 "Built successfully") | | |

### CODE REVIEW

File: Source/WebCore/Modules/model-element/ModelPlayerClient.h [hunk 1]
```diff
@@ -27,6 +27,7 @@
 
 #include "LayerHostingContextIdentifier.h"
 #include "PlatformLayerIdentifier.h"
+#include "TransformationMatrix.h"
```
Comment (mwyrzykowski) [inline Source/WebCore/Modules/model-element/ModelPlayerClient.h:30]: unified build fix

File: Source/WebCore/Modules/webaudio/OfflineAudioContext.cpp [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include "OfflineAudioContext.h"
 
 #include "AudioBuffer.h"
+#include "AudioNodeOutput.h"
```
Comment (mwyrzykowski) [inline Source/WebCore/Modules/webaudio/OfflineAudioContext.cpp]: unified build fix

File: Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp [hunk 3]
```diff
@@ -43,6 +44,7 @@
 
 #if ENABLE(MODEL_PROCESS)
 #include "ModelProcessModelPlayer.h"
+#include "ModelProcessModelPlayerManager.h"
```
Comment (mwyrzykowski) [inline Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:47]: unified build fix

File: Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp [hunk 4]
```diff
@@ -27,6 +27,7 @@
 #include "WebModelPlayerProvider.h"
 
 #include "WebPage.h"
+#include "WebProcess.h"
```
Comment (mwyrzykowski) [inline Source/WebKit/WebProcess/Model/WebModelPlayerProvider.cpp:30]: unified build fix

File: Source/WebKit/WebProcess/Plugins/WebPluginInfoProvider.cpp [hunk 5]
```diff
@@ -26,6 +26,8 @@
 #include "config.h"
 #include "WebPluginInfoProvider.h"
 
+#include <WebCore/Page.h>
```
Comment (mwyrzykowski) [inline Source/WebKit/WebProcess/Plugins/WebPluginInfoProvider.cpp:29]: unified build fix

File: Websites/webkit.org/demos/webgpu/hello-triangle-xr.html [hunk 6]
```diff
@@ -18,6 +18,7 @@
 </style>
 </head>
 <body>
+    <header/>
```
Comment (mwyrzykowski) [inline Websites/webkit.org/demos/webgpu/hello-triangle-xr.html:21]: needed for webxr-button.js

**Review by tadeuzagallo — APPROVED** [review]

**Pull request merged**